module/window_swallowing.lua

```diff
@@ -31,13 +38,16 @@ local function is_in_table(element, table)
     return res
 end
 
--- checks if parent's classname can be swallowed
-local function check_swallow(class, list)
-    if not swallowing_filter then
-        return true
-    else
-        return not is_in_table(class, list)
+-- if the swallowing filter is active checks the child and parent classes
+-- against their filters
+local function check_swallow(parent, child)
+    local res = true
+    if swallowing_filter then
+        local prnt = not is_in_table(parent, parent_filter_list)
+        local chld = not is_in_table(child, child_filter_list)
+        res = ( prnt and chld )
     end
+    return res
```

Comment:
Shouldn't the function returns false when `window_swallowing_activated` is false?

Reply:
from what i can tell no, since `window_swallowing_activated = false` means that the swallowing function will be disconected from all clients.

```lua
105 local function start()
106     client.connect_signal("manage", manage_clientspawn)
107     window_swallowing_activated = true
108 end
109
110 local function stop()
111     client.disconnect_signal("manage", manage_clientspawn)
112     window_swallowing_activated = false
113 end
114
115 local function toggle()
116     if window_swallowing_activated then
117         stop()
118     else
119         start()
120     end
121 end
```

however if you meant the boolean value of `swallowing_filter` then reffering back to the truth table the output value of this function should only be false when the filter is active and either parent or child is in any of the filtering lists.

Reply:
You're right. I have forgotten the use of his variable and thought it has what the user change to activate the module... My bad.